pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -436,9 +437,9 @@ func isTerminating(pod *v1.Pod) bool {
 	return pod.DeletionTimestamp != nil
 }
 
-// isHealthy returns true if pod is running and ready and has not been terminated
-func isHealthy(pod *v1.Pod) bool {
-	return isRunningAndReady(pod) && !isTerminating(pod)
+// isHealthy returns true if pod is running and available and has not been terminated
+func isHealthy(pod *v1.Pod, minReadySeconds int32) bool {
+	return isRunningAndAvailable(pod, minReadySeconds) && !isTerminating(pod)
```

Comment:
We probably have to make sure this change does not negatively impact other parts of the code that uses isHealthy. Would be good to track those and explicitly check. Depending on the case, would be good to have new tests there, even if it is related to other features.

Reply:
Every call to `isHealthy` is already tracked, because this PR introduces a new parameter (`minReadySeconds`) to `isHealthy`.

Every section of the code that calls `isHealthy` has a bug with a respective issue. Here's a table:


| Ocurrence(s)                                                                                                                                                                                                                                                                                                          | Issue(s)/bug(s) |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
| [L608](https://github.com/leomichalski/kubernetes/blob/25f9febd28ab56fafe248b5dd2d5d86166a9d5cc/pkg/controller/statefulset/stateful_set_control.go#L608) and [L618](https://github.com/leomichalski/kubernetes/blob/25f9febd28ab56fafe248b5dd2d5d86166a9d5cc/pkg/controller/statefulset/stateful_set_control.go#L618) | #123918   |
| [L718](https://github.com/leomichalski/kubernetes/blob/25f9febd28ab56fafe248b5dd2d5d86166a9d5cc/pkg/controller/statefulset/stateful_set_control.go#L718)                                                                                                                                                              | #119234   |
| [L762](https://github.com/leomichalski/kubernetes/blob/25f9febd28ab56fafe248b5dd2d5d86166a9d5cc/pkg/controller/statefulset/stateful_set_control.go#L762)                                                                                                                                                              | #119234  and #112307 |

We really need more tests.